### PR TITLE
Top Down RPG Walking Simulator

### PR DESCRIPTION
**Pull request by colton-smith**
## Brief
Initial PR for the top-down RPG demo. Currently, there is a plain map with an animated sprite that can walk around. I will customize the map as much as I can throughout today to make it look prettier than concrete (maybe add a road and some street lights or something) 

## Summary
- `Level` class which encapsulates drawing a WxH grid of specified textures from a spritesheet
- `LevelEditorLayer` class which really should be called RPGLayer or something, but I kinda started out this branch thinking I was gonna make a way to create a level
- `Player` class which is just hard coded player movement and sprite animation. Its cool how it works and it would be fun to generalize the sprite animation so it could be easily used for anything (trees blowing in the wind)

### CODE REVIEW

File: Sandbox/RPG/Level.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#include "Aquarius.h"
+
+#include <array>
+#include <string> 
+
```
Comment (liamjwarren) [inline Sandbox/RPG/Level.h:5]: Add line

File: Sandbox/RPG/LevelEditorLayer.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
+#include "LevelEditorLayer.h"
+
```
Comment (liamjwarren) [inline Sandbox/RPG/LevelEditorLayer.cpp:2]: Add line

File: Sandbox/RPG/Player.h [hunk 3]
```diff
@@ -0,0 +1,192 @@
+#pragma once 
+
+#include "Aquarius.h"
+#include <chrono>
+
```
Comment (liamjwarren) [inline Sandbox/RPG/Player.h:5]: Add line

File: Sandbox/RPG/Player.h [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+	std::vector<Aquarius::QuadTexCoords> m_walkRightFrames;
+	std::vector<Aquarius::QuadTexCoords> m_walkUpFrames;
+	std::vector<Aquarius::QuadTexCoords> m_walkDownFrames;
+	std::vector<Aquarius::QuadTexCoords> m_idlingFrames;
+	std::vector<Aquarius::QuadTexCoords>* m_activeFrames;
+	int m_offset;
+
+	Aquarius::timeDelta_t m_count;
+	Aquarius::timeDelta_t m_walkInterval;
+};
```
Comment (liamjwarren) [inline Sandbox/RPG/Player.h]: add line EOF

**Review by Burke-Daniel — APPROVED** [review]
GTM

**Pull request merged**